internal/security/bootstrapper/messagebus/handlers/handlers.go

```diff
@@ -0,0 +1,181 @@
+			break
+		}
+
+		lc.Warnf("Could not retrieve message bus credentials (startup timer has not expired): %s", err.Error())
+		startupTimer.SleepForInterval()
+	}
+
+	if credentials.Password == "unset" {
+		lc.Error("Failed to retrieve message bus credentials before startup timer expired")
+		return false
+	}
+
+	handler.credentials = credentials
+	return true
+}
+
+// SetupPasswordFile sets up the message bus password file for MQTT specific
+func (handler *Handler) SetupPasswordFile(ctx context.Context, _ *sync.WaitGroup, startupTimer startup.Timer,
+	dic *di.Container) bool {
+	lc := bootstrapContainer.LoggingClientFrom(dic.Get)
+	config := container.ConfigurationFrom(dic.Get)
+
+	if config.MessageQueue.Type != "mqtt" {
+		lc.Infof("messaage bus not mqtt type: %s, skip setting up password file", config.MessageQueue.Type)
+		return true
+	}
+
+	pwdFile := config.MessageQueue.Optional["PasswordFile"]
+	if len(pwdFile) == 0 {
+		lc.Errorf("missing PasswordFile configuration for mqtt")
+		return false
+	}
+
+	pwdFileBaseDir := filepath.Dir(pwdFile)
+	if err := os.Chdir(pwdFileBaseDir); err != nil {
+		lc.Errorf("failed to change directory %s: %v", pwdFileBaseDir, err)
+		return false
+	}
+
+	// use mqtt password utility to set up password file
+	cmd := exec.Command("mosquitto_passwd", "-c", "-b", pwdFile, handler.credentials.Username, handler.credentials.Password)
+	_, err := cmd.Output()
+
+	if err != nil {
+		lc.Errorf("failed to execute command mosquitto_passwd: %v", err)
+		return false
+	}
+
+	return true
+}
+
+// SetupConfFile dynamically creates message bus config file
+func (handler *Handler) SetupConfFile(ctx context.Context, _ *sync.WaitGroup, _ startup.Timer,
+	dic *di.Container) bool {
+	lc := bootstrapContainer.LoggingClientFrom(dic.Get)
+	config := container.ConfigurationFrom(dic.Get)
+
+	if config.MessageQueue.Type != "mqtt" {
+		lc.Infof("message bus not mqtt type: %s, skip setting up config file", config.MessageQueue.Type)
+		return true
+	}
+
+	brokerConfigFile := config.MessageQueue.Optional["BrokerConfigFile"]
+	if len(brokerConfigFile) == 0 {
+		lc.Errorf("missing brokerConfigFile configuration for mqtt")
+		return false
+	}
+
+	configFileTemplate := `listener {{.MQTTPort}}
+password_file {{.PwdFilePath}}`
+
+	type mqttConfig struct {
+		MQTTPort    int
+		PwdFilePath string
+	}
+
+	mqttConf, err := template.New("mqtt-config").Parse(configFileTemplate + fmt.Sprintln())
+	if err != nil {
+		lc.Errorf("failed to parse MQTT config file template %s: %v", configFileTemplate, err)
+		return false
+	}
+
+	lc.Infof("Creating the broker config file: %s", brokerConfigFile)
+
+	// open config file with read-write and overwritten attribute (TRUNC)
+	confFile, err := os.OpenFile(brokerConfigFile, os.O_RDWR|os.O_CREATE|os.O_TRUNC, readWriteOnlyForOwner)
+	if err != nil {
+		lc.Errorf("failed to open config file %s: %v", brokerConfigFile, err)
+		return false
+	}
+	defer func() {
+		_ = confFile.Close()
+	}()
```

Comment:
I think the preferred style for writing files is to drop the Flush() and just do the Close() inline and at anytime you bail early.

Reply:
this is idiomatic way for `file open` to defer file close() right after.  we use flush because we use buffer writer